emission/storage/timeseries/non_user_timeseries.py

```diff
@@ -51,16 +53,28 @@ def insert(self, entry):
         Inserts the specified entry and returns the object ID 
         """
         logging.debug("insert called")
-        if type(entry) == dict:
+        if isinstance(entry, dict):
             entry = ecwe.Entry(entry)
         if entry["user_id"] is not None:
-            raise AttributeError("Saving entry %s for %s in non_user_timeseries" % 
-                (entry, entry["user_id"]))
+            raise AttributeError(
+                f"Saving entry {entry} for {entry['user_id']} in non_user_timeseries is not allowed."
+            )
         else:
             logging.debug("entry was fine, no need to fix it")
 
-        logging.debug("Inserting entry %s into timeseries" % entry)
-        return self.get_timeseries_db(entry.metadata.key).insert(entry)
+        # Get the collection and log its full name
+        collection = self.get_timeseries_db(entry.metadata.key)
+        logging.debug(f"Collection used for insertion: {collection.full_name}")
+
+        logging.debug(f"Inserting entry {entry} into timeseries")
+        try:
+            result = collection.insert_one(entry)
+            logging.debug(f"Inserted entry with ID: {result.inserted_id}")
+            return result.inserted_id
+        except pymongo.errors.PyMongoError as e:
+            logging.error(f"Failed to insert entry: {e}")
+            raise
+
```

Comment:
I don't think this is wrong, but can you highlight why this was needed?

Reply:
I didn't know where the current tests were being stored so included this to track it down; hence where I found it in aggregate